feat(template): use environment variables in templates (#19301)
renovatebot/renovate: docs/usage/templates.md

Returns `true` if at least one expression is `true`.

`{{#if (or isPatch isSingleVersion}}Small update, safer to merge and release.{{else}}Check out the changelog for all versions before merging!{{/if}}`

## Environment variables

> For security reasons, you can only access a handful of basic environment variables like `HOME` or `PATH` by default:

Reviewer: We should mention `exposeAllEnv`[^1] here to allow all env. @HonkingGoose Can you help?

[^1]: https://docs.renovatebot.com/self-hosted-configuration/#exposeallenv

> `HOME is {{HOME}}`

Author: 1. When spaces are added inside `{{ env.HOME }}`, regex `templateRegex` does not extract the variable name. Are you sure they should be there?
2. Prefix `env.` implemented.

Returns `true` if at least one expression is `true`.

`{{#if (or isPatch isSingleVersion}}Small update, safer to merge and release.{{else}}Check out the changelog for all versions before merging!{{/if}}`

## Environment variables

> For security reasons, you can only access a handful of basic environment variables like `HOME` or `PATH` by default:

Reviewer 2: ```suggestion
By default, you can only access a handful of basic environment variables like `HOME` or `PATH`.
This is for security reasons.
```

`HOME is {{env.HOME}}`

> If self-hosting, you can expose additional variables with setting [`customEnvVariables`](https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables).

Reviewer 2: ```suggestion
If you're self-hosting Renovate, you can expose additional variables with the  [`customEnvVariables`](https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables) config option.
```

> You can alse use setting [`exposeAllEnv`](https://docs.renovatebot.com/self-hosted-configuration/#exposeallenv) to allow
> using all environment variables in templates, but always consider the security implications of using that setting.

Reviewer 2: ```suggestion
You can also use the [`exposeAllEnv`](https://docs.renovatebot.com/self-hosted-configuration/#exposeallenv) config option to allow all environment variables in templates.
Always consider the security implications of using `exposeAllEnv`.
```

> You can also use the [`exposeAllEnv`](https://docs.renovatebot.com/self-hosted-configuration/#exposeallenv) config
> option to allow all environment variables in templates, but make sure to consider the security implications of
> giving the scripts unrestricted access to all variables.

Reviewer 3: ```suggestion
You can also use the [`exposeAllEnv`](https://docs.renovatebot.com/self-hosted-configuration/#exposeallenv) config option to allow all environment variables in templates, but make sure to consider the security implications of giving the scripts unrestricted access to all variables.
```

Single line per sentence

> If secrets or any other confidential information is stored in environment variables, they could be leaked
> by a malicious script.

Reviewer 2: ```suggestion
Secrets and other confidential information stored in environment variables could be leaked by a malicious script.
```

> Secrets and other confidential information stored in environment variables could be leaked by a malicious script.

Reviewer: @HonkingGoose @rarkins should we add a bigger warning here?

Reviewer 2: We should use a warning admonition to grab the readers attention.

I think the key points to cover are:

- Explain what's the benefit of enabling the dangerous option
- Explain why it can be dangerous
- If there's any way to make it less dangerous, explain how
- Give our recommendations as maintainers
- Warn loudly about dangerous stuff

Reviewer: can you make a suggestion?

Reviewer 3: I think exposeAllEnv should have the big warning while this docs section just references it

Author: I added the big warning to `docs/usage/self-hosted-configuration.md` and left only a short remark in `docs/usage/templates.md`.
